lib/virtio/virtqueue.c

```diff
@@ -184,6 +261,10 @@ void *virtqueue_get_buffer(struct virtqueue *vq, uint32_t *len, uint16_t *idx)
 
 	atomic_thread_fence(memory_order_seq_cst);
 
+#ifdef VIRTIO_D_CACHE
```

Comment:
why doing it twice time?

Reply:
yeah, I was not 100% sure about this and now thinking about this again it looks wrong indeed. I added it twice because of the synchronization fence `atomic_thread_fence` but that's valid only on the same CPU so the second invalidation is not needed.